### ISSUE TITLE
downscale-period not working as expected

### ISSUE THREAD

**Issue by benraven** [hjacobs/kube-downscaler#65]
My understanding of the documentation is that by annotating a deployment `downscaler/downscale-period="Mon-Sun 19:00-19:30 UTC"`, it should scale down to zero and never scale back up again. In practice, the downscaler simply ignores this deployment, as if it had no annotation at all. Am I misunderstanding the documentation?

Additionally, it would be extremely helpful if the Usage section could have a couple of examples for each command line option and annotation, and a note of their limitations. For example, I assumed that the `--namespace` option could take more than one namespace as a value, as the default is all namespaces, but I can't find a way of formatting a list of namespaces that it understands, making me think it's either all or just one, like kubectl commands. I would be more than happy to write that extra documentation, as soon as I fully understand it myself.

**Comment by hjacobs** [comment]
Good point about `--namespace`: I added a sentence about the scenario it was designed for: https://github.com/hjacobs/kube-downscaler/commit/c176eadd803e8a75f0fa4ea7c6dbc3da65fd733d

TL;DR: The usual use is to run `kube-downscaler` for the whole cluster and only exclude namespaces/resources.

Regarding `downscale-period`: what options did you specify and what other annotations did you add?

**Comment by benraven** [comment]
Thanks for the quick response, and the clarification!

That `downscale-period` annotation is the only one on the deployments, with no annotations on the namespace, and the only options I specified for the kube-downscaler deployment itself are `--interval`, `--exclude-namespaces` (not including the namespace that the deployments I tested are in) and `--debug`.

The annotation we use on everything else is `downscaler/downtime: "Mon-Fri 00:00-07:00 Europe/London,Mon-Fri 19:30-24:00 Europe/London,Sat-Sun 00:00-24:00 Europe/London"` which works perfectly. We use that three-part downtime instead of just `downscaler/uptime: "Mon-Fri 07:00-19:30 Europe/London"` because we want things that have been manually scaled down to stay scaled down until manually scaled up, and that seemed like the best way of doing that. We'd also like to be able to use `downscale-period` to indefinitely scale down certain things that have been manually scaled up, so that unused dev instances don't linger forever. Again, I hope I'm understanding the documentation correctly and that those are the right ways to achieve our goals.

**Comment by benraven** [comment]
#70 fixes the remaining issue I had regarding the downscaler ignoring annotations on deployments in namespaces that don't have annotations.

Unfortunately, the change to what is now [Line 87](https://github.com/hjacobs/kube-downscaler/blob/4044f52dbf02f17eb6acbe7bde1df1c595ef1c23/kube_downscaler/scaler.py#L87) results in an error about is_uptime being referenced before declaration, but it still works.
